Curve-finance-suite/Unkill/src/agent.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Finding,
+  HandleTransaction,
+  FindingSeverity,
+  FindingType,
+} from "forta-agent";
+import { provideFunctionCallsDetectorHandler } from "forta-agent-tools";
+import { stablePoolInterface } from "./abi";
+
+const CURVE_POOL_OWNER = "0xeCb456EA5365865EbAb8a2661B0c503410e9B347";
+
+export const createFinding = (functionCallInfo: any): Finding => {
+  return Finding.fromObject({
+    name: "UnkillMe Agent",
+    description: "Pool Owner called unkill_me method",
+    alertId: "CURVE-12",
+    type: FindingType.Info,
+    severity: FindingSeverity.Info,
+    protocol: "Curve Finance",
+    metadata: {
+      UnkillPool: functionCallInfo.to,
+    },
+  });
+};
+
+export const provideHandleTransaction = (
+  poolOwnerAddress: string
+): HandleTransaction =>
+  provideFunctionCallsDetectorHandler(
+    createFinding,
+    stablePoolInterface.getFunction("unkill_me").format(),
+    { from: poolOwnerAddress }
+  );
+
+export default {
+  handleTransaction: provideHandleTransaction(CURVE_POOL_OWNER),
```

Comment:
From my experience it was always standard to have the `alertId` as an input here so that the `alertId` could be changed by whoever is using the agent if necessary

Reply:
Yes that is one old standard we use in agent templates, not needed when creating common agents template.

Reply:
Actually using `alertId` as an argument was a introduced when creating `templates` as @e1Ru1o mentioned. For regular agents I think is cleaner to write them like this.